From 51d136ba44f338be9cf7105d91c3882638f3f323 Mon Sep 17 00:00:00 2001
From: Gary Bisson <gary.bisson@boundarydevices.com>
Date: Mon, 14 Dec 2020 14:26:01 +0100
Subject: [PATCH] start_isp.sh: fix test to be generic

- so that it works on EVK and any other i.MX 8M Plus platform
- otheriwse misses sensors behind an i2c multiplexer

Signed-off-by: Gary Bisson <gary.bisson@boundarydevices.com>
---
 imx/start_isp.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/imx/start_isp.sh b/imx/start_isp.sh
index 6daabc59c..892b7c061 100755
--- a/imx/start_isp.sh
+++ b/imx/start_isp.sh
@@ -7,8 +7,8 @@
 #
 
 RUNTIME_DIR="$( cd "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
-NR_DEVICE_TREE_BASLER=$(grep basler-camera-vvcam /sys/firmware/devicetree/base/soc@0/*/i2c@*/*/compatible -l | wc -l 2> /dev/null)
-NR_DEVICE_TREE_OV5640=$(grep ov5640 /sys/firmware/devicetree/base/soc@0/*/i2c@*/*/compatible -l | wc -l 2> /dev/null)
+NR_DEVICE_TREE_BASLER=$(grep basler-camera-vvcam `find /sys/firmware/devicetree/base/soc@0/ -name compatible | grep i2c` -l | wc -l 2> /dev/null)
+NR_DEVICE_TREE_OV5640=$(grep ov5640 `find /sys/firmware/devicetree/base/soc@0/ -name compatible | grep i2c` -l | wc -l 2> /dev/null)
 
 # check if the basler device has been enabled in the device tree
 if [ $NR_DEVICE_TREE_BASLER -eq 1 ]; then
-- 
2.33.0

